tests/test_cached_property.py

```diff
@@ -77,31 +122,39 @@ def test_cached_property(self):
         self.assert_control(check, 2)
 
         # The cached version demonstrates how nothing is added after the first
-        self.assert_cached(check, 1)
-        self.assert_cached(check, 1)
+        for cached_attr, total_attr in self.cached_attrs:
```

Comment:
I was unsure if this was the right thing to do as well. It makes the tests a bit more complicated. I considered making a separate TestCase classes for each type of attribute, but considering that the duplicate tests would be testing pretty much the same thing, it made sense to me to test all types of attribute in a single test.